packages/Popover/src/Popover.js

```diff
@@ -0,0 +1,500 @@
+
+    justifications[justify].some(candidate => {
+      if (!this.checkJustification(candidate)) {
+        return false;
+      }
+
+      validJustification = candidate;
+      return true;
+    });
+
+    return validJustification;
+  }
+
+  // Checks that an alignment will not cause the popover to collide with the window.
+  checkAlignment(alignment) {
+    const { withoutPortal } = this.props;
+    const top = withoutPortal
+      ? this.calcWithoutPortalTop({ alignment })
+      : this.calcTop({ alignment });
+    const left = withoutPortal
+      ? this.calcWithoutPortalLeft({ alignment })
+      : this.calcLeft({ alignment });
+
+    if (['top', 'bottom'].includes(alignment)) {
+      return this.checkVerticalWindowCollision(top);
+    }
+
+    if (['left', 'right'].includes(alignment)) {
+      return this.checkHorizontalWindowCollision(left);
+    }
+  }
+
+  // Checks that a justification will not cause the popover to collide with the window.
+  checkJustification(justification) {
+    const { withoutPortal } = this.props;
+    const top = withoutPortal
+      ? this.calcWithoutPortalTop({ justification })
+      : this.calcTop({ justification });
+    const left = withoutPortal
+      ? this.calcWithoutPortalLeft({ justification })
+      : this.calcLeft({ justification });
+
+    if (['top', 'bottom', 'center-vertical'].includes(justification)) {
+      return this.checkVerticalWindowCollision(top);
+    }
+
+    if (['left', 'right', 'center-horizontal'].includes(justification)) {
+      return this.checkHorizontalWindowCollision(left);
+    }
+  }
+
+  // Check if horizontal position collides with edge of window
+  checkHorizontalWindowCollision(left) {
+    const contentRefPos = this.getRefPosition(this.contentRef)
+    const tooWide = left + contentRefPos.width > this.state.windowWidth;
+
+    return !(left < 0 || tooWide);
+  }
+
+  // Check if vertical position collides with edge of window
+  checkVerticalWindowCollision(top) {
+    const contentRefPos = this.getRefPosition(this.contentRef);
+    const tooTall = top + contentRefPos.height > this.state.windowHeight;
+
+    return !(top < 0 || tooTall);
+  }
+  ////////////////////////////////////////////////////////////////////////////////////////////
+
+  // Functions to calculate top and left positioning
+  // Based on whether or not a Portal is being leveraged
+
+  // Returns the 'top' position in pixels for a valid alignment or justification.
+  calcTop({ alignment, justification }) {
+    const { referenceElPos } = this.state;
+    const contentRefPos = this.getRefPosition(this.contentRef);
+
+    if (alignment) {
+      switch (alignment) {
+        case 'top':
+          return referenceElPos.top - contentRefPos.height;
+
+        case 'bottom':
+          return referenceElPos.top + referenceElPos.height;
+      }
+    }
+
+    if (justification) {
+      switch (justification) {
+        case 'top':
+          return referenceElPos.top;
+        case 'bottom':
```

Comment:
Nit: line break between these case blocks, especially given we aren't using `{}` here.

Reply:
👍